src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/IdentifierFilter.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Filter;
+
+use Pim\Component\Catalog\Exception\InvalidOperatorException;
+use Pim\Component\Catalog\Query\Filter\FieldFilterHelper;
+use Pim\Component\Catalog\Query\Filter\FieldFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+use Pim\Component\Catalog\Repository\AttributeRepositoryInterface;
+
+/**
+ * Identifier filter for an Elasticsearch query
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IdentifierFilter extends AbstractFieldFilter implements FieldFilterInterface
+{
+    const IDENTIFIER_KEY = 'identifier';
+
+    /** @var AttributeRepositoryInterface */
+    protected $attributeRepository;
+
+    /**
+     * @param AttributeRepositoryInterface $attributeRepository
+     * @param array                        $supportedFields
+     * @param array                        $supportedOperators
+     */
+    public function __construct(
+        AttributeRepositoryInterface $attributeRepository,
+        array $supportedFields = [],
+        array $supportedOperators = []
+    ) {
+        $this->supportedFields = $supportedFields;
+        $this->supportedOperators = $supportedOperators;
+        $this->attributeRepository = $attributeRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFieldFilter($field, $operator, $value, $locale = null, $scope = null, $options = [])
+    {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkValue($field, $operator, $value);
+
+        switch ($operator) {
+            case Operators::STARTS_WITH:
+                $clause = [
+                    'query_string' => [
+                        'default_field' => self::IDENTIFIER_KEY,
```

Comment:
Nope, I agree with what is done here, we should always use `identifier`.
In the future, it is possible that identifier will not even remains a product value.